app/controllers/chat_channels_controller.rb

```diff
@@ -112,6 +112,18 @@ def open_chat
     end
   end
 
+  def block_chat
+    chat_channel = ChatChannel.find(params[:chat_id])
+    policy = ChatChannelPolicy.new(current_user, chat_channel)
+    if policy.open? && chat_channel.valid? && chat_channel.channel_type == "direct"
```

Comment:
This isn't really how policies are meant to be used, but this is technically solving the problem. Since this is outside of where you typically contribute, I'll just merge this and refactor it myself later and let you know what I did. I think that's simpler than holding this up right now.